rs/replicated_state/src/canister_state/system_state.rs

```diff
@@ -1451,6 +1451,62 @@ impl SystemState {
             .time_out_messages(current_time, own_canister_id, local_canisters)
     }
 
+    /// Enqueues "deadline expired" references for all expired best-effort callbacks
+    /// without a response.
+    ///
+    /// Returns `Err` if a `SystemState` internal inconsistency prevented  one or
+    /// more "deadline expired" references from being enqueued.
+    pub fn time_out_callbacks(
+        &mut self,
+        current_time: CoarseTime,
+        own_canister_id: &CanisterId,
+        local_canisters: &BTreeMap<CanisterId, CanisterState>,
+    ) -> Result<(), StateError> {
```

Comment:
Do we really still want to return a state error here? If we'd introduce some custom error type that can accumulate the failed callbacks we could even log some information that might be relevant for a manual recovery in such a case.

Reply:
I could return a `Vec<StateError>`, if we're concerned about failing to enqueue multiple callbacks and want to rely on the actual error message to recover them. But I believe the details in `StateError::NonMatchingResponse`(why we failed to enqueue it, the respondent, the deadline) are actually useful for debugging the root cause (is this a subnet-local message? was the queue already gone? etc.).